public/app/plugins/datasource/cloudwatch/components/VariableQueryEditor/VariableQueryEditor.tsx

```diff
@@ -126,13 +131,17 @@ export const VariableQueryEditor = ({ query, datasource, onChange }: Props) => {
             onChange={(value: string) => onQueryChange({ ...parsedQuery, dimensionKey: value })}
             label="Dimension Key"
           />
-          <VariableTextField
-            value={query.dimensionFilters}
-            tooltip='A JSON object representing dimensions and the values to filter on. Ex. { "filter_name1": [ "filter_value1" ], "filter_name2": [ "*" ] }'
-            placeholder='{"key":["value"]}'
-            onBlur={(value: string) => onQueryChange({ ...parsedQuery, dimensionFilters: value })}
-            label="Filters"
-          />
+          <InlineField label="Dimensions" labelWidth={20} tooltip="Dimensions to filter the returned values on">
+            <Dimensions
+              query={{ ...parsedQuery, dimensions: parsedQuery.valueDimensions }}
+              onChange={(dimensions) => {
+                onChange({ ...parsedQuery, valueDimensions: dimensions });
+              }}
+              dimensionKeys={valueDimensionKeys}
+              disableExpressions={true}
+              datasource={datasource}
+            />
+          </InlineField>
```

Comment:
Do you mean tests for the `Dimensions` element itself, or tests for it as part of this form? The `Dimensions` element has tests (that I forgot to move in the first commit), but I can add new tests if you mean the second one.

Reply:
I mean this component. There is a bit of logic in place here like parsing the query and setting `valueDimensions` and `valueDimensionKeys` or checking that `onChange` is called with the right thing.

Also, unrelated to this (and probably unrelated to this PR), passing down `datasource` should be generally avoided. That's a complex object that makes it more difficult to test components. It's usually better to pass down just the methods that are needed (not sure what's the thing needed here).

Reply:
Added a test that changes that parses a filter and then changes the key and value.
I'll file a ticket to stop passing down datasource after this gets merged (since it moves the dimensions files).